plutus-core/testlib/PlutusCore/Generators/QuickCheck/Builtin.hs

```diff
@@ -55,44 +56,69 @@ The 'Arbitrary' instances for 'Integer' and 'Int64' only generate small integers
     >>> fmap (any ((> 30) . abs) . concat . concat . concat) . sample' $ arbitrary @[[[Int]]]
     False
 
-We want to at least occasionally generate some larger ones, which is what the 'Arbitrary'
-instance for 'Int64' does:
+We want to generate larger ones, including converted-to-Integer 'minBound' and 'maxBound' of various
+integral types. Hence we declare 'nextInterestingBound' and 'highInterestingBound' to specify the
+"interesting" ranges to generate positive integers within. We also make it likely to hit either end
+of each of those ranges.
+-}
 
-    >>> import Data.Int
-    >>> fmap (any ((> 10000) . abs) . concat . concat . concat) . sample' $ arbitrary @[[[Int64]]]
-    True
+-- See Note [QuickCheck and integral types].
+nextInterestingBound :: Integer -> Integer
+nextInterestingBound 1 = 127
+nextInterestingBound x = (x + 1) ^ (2 :: Int) * 2 - 1
```

Comment:
This kind of works accidentally because of the way the sizes of the fixed-width integral types happen to grow. I was going to suggest having a function with an explicit case for each width, but having thought about it that would probably be really horrible.

Reply:
> This kind of works accidentally because of the way the sizes of the fixed-width integral types happen to grow.

Probably not gonna change anytime soon! I'm keeping it as is for now, we can always change it later.